src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfExonFeature.java

```diff
@@ -0,0 +1,145 @@
+                                           GenomicStrand genomicStrand,
+                                           GenomicPhase genomicPhase,
+                                           String geneId,
+                                           String transcriptId,
+                                           GeneTranscriptType geneType,
+                                           GeneTranscriptStatus geneStatus,
+                                           String geneName,
+                                           GeneTranscriptType transcriptType,
+                                           GeneTranscriptStatus transcriptStatus,
+                                           String transcriptName,
+                                           int exonNumber,
+                                           String exonId,
+                                           LocusLevel locusLevel,
+                                           ArrayList<OptionalField<?>> optionalFields,
+                                           String anonymousOptionalFields) {
+
+        return new GencodeGtfExonFeature(featureOrderNumber, chromosomeName, annotationSource, featureType, genomicStartLocation, genomicEndLocation, genomicStrand, genomicPhase, geneId, transcriptId, geneType, geneStatus, geneName, transcriptType, transcriptStatus, transcriptName, exonNumber, exonId, locusLevel, optionalFields, anonymousOptionalFields);
+    }
+
+    // ============================================================================================================
+
+
+    GencodeGtfCDSFeature                        cds = null;
+    GencodeGtfStartCodonFeature                 startCodon = null;
+    GencodeGtfStopCodonFeature                  stopCodon = null;
+
+    // ============================================================================================================
+
+    public GencodeGtfCDSFeature getCds() {
+        return cds;
+    }
+
+    public GencodeGtfStartCodonFeature getStartCodon() {
+        return startCodon;
+    }
+
+    public GencodeGtfStopCodonFeature getStopCodon() {
+        return stopCodon;
+    }
+
+    // ============================================================================================================
+
+    public void setCds(GencodeGtfCDSFeature cds) {
+        this.cds = cds;
+    }
+
+    public void setStartCodon(GencodeGtfStartCodonFeature startCodon) {
+        this.startCodon = startCodon;
+    }
+
+    public void setStopCodon(GencodeGtfStopCodonFeature stopCodon) {
+        this.stopCodon = stopCodon;
+    }
+
+
+    @Override
+    protected List<GencodeGtfFeature> getAllFeatures() {
+        ArrayList<GencodeGtfFeature> list = new ArrayList<>();
+        list.add(this);
+
+        if ( cds != null ) { list.add(cds) ; }
+        if ( startCodon != null ) { list.add(startCodon) ; }
+        if ( stopCodon != null ) { list.add(stopCodon) ; }
+
+        return list;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
When you override `equals()`, you should always override `hashCode()` as well, and ensure that they are mutually consistent. IntelliJ's auto-generated versions of these methods are usually a good starting point, though you often want `instanceof` instead of `getClass()` when dealing with a class hierarchy, as I mentioned elsewhere in this review.